lib/galaxy_test/api/test_libraries.py

```diff
@@ -110,7 +110,7 @@ def test_get_library_available_permissions(self):
         library_id = library["id"]
         role_id = self.library_populator.user_private_role_id()
         # As we can manage this library our role will be available
-        available = self.library_populator.get_permissions(library_id, scope="available")
+        available = self.library_populator.get_permissions(library_id, scope="available", page_limit=1000)
```

Comment:
I think this still won't set pagination because `page` is `None`
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy_test/base/populators.py#L1240  (you have a conjunction in this line, so you need both page and page_limit for pagination to be set)

In that regard, I wonder if it might be a good idea to add an assertion to the top of the `get_permissions` method (negated XOR?) to enforce that we either provide both values or none? Otherwise we'll be passing one or the other and then hunting down where one got lost.

Unless I'm misreading it all...

Reply:
You are absolutely right!
I have reverted the changes and just increase the default `page_limit` for all tests. Third time lucky :sweat_smile: